### PR TITLE
JAMES-2506 remove easymock jmock

### PR DESCRIPTION
**Pull request by trantienduchn**
detached from https://github.com/linagora/james-project/pull/1618

wait for https://github.com/linagora/james-project/pull/1618 being merged, then rebase master, then trigger CI build

### CODE REVIEW

File: protocols/imap/src/test/java/org/apache/james/imap/decode/parser/SearchCommandParserCharsetTest.java [hunk 1]
```diff
@@ -80,29 +77,27 @@
...
-        context.checking(new Expectations() {{
-                oneOf(mockStatusResponseFactory).taggedNo(
-                        with(equal(TAG)), 
-                        with(same(command)), 
-                        with(equal(HumanReadableText.BAD_CHARSET)),
-                        with(equal(StatusResponse.ResponseCode.badCharset(CharsetUtil.getAvailableCharsetNames()))));
-                }
-            }
-        );
+    public void testBadCharsetMockito() throws Exception {
```
Comment (mbaechler) [inline protocols/imap/src/test/java/org/apache/james/imap/decode/parser/SearchCommandParserCharsetTest.java]: please remove Mockito suffix

File: server/container/cli/src/test/java/org/apache/james/cli/ServerCmdTest.java [hunk 2]
```diff
@@ -1187,12 +1007,10 @@ public void createMailboxMappingCommandShouldThrowOnAdditionalArguments() throws
         String[] arguments = { "-h", "127.0.0.1", "-p", "9999", CmdType.CREATEMAILBOX.getCommand(), namespace, user, name, ADDITIONAL_ARGUMENT };
         CommandLine commandLine = ServerCmd.parseCommandLine(arguments);
 
-        control.replay();
         try {
             testee.executeCommandLine(commandLine);
         } finally {
-            control.verify();
-        }
+            }
```
Comment (mbaechler) [inline server/container/cli/src/test/java/org/apache/james/cli/ServerCmdTest.java]: why do you keep the finally blocks ?

File: server/container/cli/src/test/java/org/apache/james/cli/ServerCmdTest.java [hunk 3]
```diff
@@ -865,25 +729,21 @@ public void copyMailboxMappingCommandShouldThrowOnMissingArguments() throws Exce
...
     @Test(expected = InvalidArgumentNumberException.class)
     public void deleteUserMailboxesMappingCommandShouldThrowOnMissingArguments() throws Exception {
         String[] arguments = { "-h", "127.0.0.1", "-p", "9999", CmdType.DELETEUSERMAILBOXES.getCommand()};
         CommandLine commandLine = ServerCmd.parseCommandLine(arguments);
 
-        control.replay();
         try {
             testee.executeCommandLine(commandLine);
         } finally {
-            control.verify();
```
Comment (mbaechler) [inline server/container/cli/src/test/java/org/apache/james/cli/ServerCmdTest.java]: you dropped the verify stage ?

File: server/container/cli/src/test/java/org/apache/james/cli/ServerCmdTest.java (hunk 2 above)
Comment (trantienduchn) [reply server/container/cli/src/test/java/org/apache/james/cli/ServerCmdTest.java]: my mistake, i used find & replace text to replace Easymock by mockito

File: server/container/cli/src/test/java/org/apache/james/cli/ServerCmdTest.java (hunk 3 above)
Comment (trantienduchn) [reply server/container/cli/src/test/java/org/apache/james/cli/ServerCmdTest.java]: in some tests there is no expectation, and I think I can remove the `control.verify()` which is used to verify expectations.

**Comment by trantienduchn** [comment]
mockito upgrade problem, faced before, wait for mockito merged then rebase master 

```
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Tests run: 19, Failures: 5, Errors: 0, Skipped: 0, Time elapsed: 0.75 sec <<< FAILURE! - in org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] containsDomainShouldReturnDetectedDomains  Time elapsed: 0.012 sec  <<< FAILURE!
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] org.junit.ComparisonFailure: expected:<[tru]e> but was:<[fals]e>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedDomains(AbstractDomainListPrivateMethodsTest.java:294)
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] getDomainsShouldContainDetectedDomainsAndIpsOfAddedDomains  Time elapsed: 0.034 sec  <<< FAILURE!
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] java.lang.AssertionError: 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Expecting:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <["localhost", "495028a55458", "added.tld", "148.25.32.2"]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] to contain:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <["detected.tld", "148.25.32.1", "added.tld", "148.25.32.2"]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] but could not find:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <["detected.tld", "148.25.32.1"]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomainsAndIpsOfAddedDomains(AbstractDomainListPrivateMethodsTest.java:216)
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] containsDomainShouldReturnDetectedIp  Time elapsed: 0.001 sec  <<< FAILURE!
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] org.junit.ComparisonFailure: expected:<[tru]e> but was:<[fals]e>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedIp(AbstractDomainListPrivateMethodsTest.java:243)
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] getDomainsShouldContainDetectedDomains  Time elapsed: 0.002 sec  <<< FAILURE!
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] java.lang.AssertionError: 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Expecting:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : localhost, Domain : 495028a55458]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] to contain:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : detected.tld]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] but could not find:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : detected.tld]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomains(AbstractDomainListPrivateMethodsTest.java:172)
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] getDomainsShouldContainDetectedDomainsAndIps  Time elapsed: 0.001 sec  <<< FAILURE!
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] java.lang.AssertionError: 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Expecting:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : localhost, Domain : 495028a55458]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] to contain:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : detected.tld, Domain : 148.25.32.1]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] but could not find:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : detected.tld, Domain : 148.25.32.1]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 	at org.apache.james.domainlist.lib.AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomainsAndIps(AbstractDomainListPrivateMethodsTest.java:190)
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Results :
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Failed tests: 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]   AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedDomains:294 expected:<[tru]e> but was:<[fals]e>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]   AbstractDomainListPrivateMethodsTest.containsDomainShouldReturnDetectedIp:243 expected:<[tru]e> but was:<[fals]e>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]   AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomains:172 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Expecting:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : localhost, Domain : 495028a55458]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] to contain:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : detected.tld]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] but could not find:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : detected.tld]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]   AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomainsAndIps:190 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Expecting:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : localhost, Domain : 495028a55458]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] to contain:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : detected.tld, Domain : 148.25.32.1]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] but could not find:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <[Domain : detected.tld, Domain : 148.25.32.1]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]   AbstractDomainListPrivateMethodsTest.getDomainsShouldContainDetectedDomainsAndIpsOfAddedDomains:216 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Expecting:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <["localhost", "495028a55458", "added.tld", "148.25.32.2"]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] to contain:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <["detected.tld", "148.25.32.1", "added.tld", "148.25.32.2"]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] but could not find:
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af]  <["detected.tld", "148.25.32.1"]>
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] 
[4e865e07349e1a1e6d949515b9ba5f1082eaf9af] Tests run: 82, Failures: 5, Errors: 0, Skipped: 0
```

**Comment by trantienduchn** [comment]
lucene test fails again :) have to wait for lucene fix
```
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Tests in error: 
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   Search.testSearchAtomsITALY:63 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apache/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   Search.testSearchAtomsKOREA:71 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apache/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   Search.testSearchAtomsUS:55 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apache/jam...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   Search.testSearchCombinationsITALY:85 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   Search.testSearchCombinationsKOREA:92 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   Search.testSearchCombinationsUS:78 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apa...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   SelectedState.testSearchITALY:130 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apac...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   SelectedState.testSearchKOREA:193 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apac...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   SelectedState.testSearchUS:67 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apache/j...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   SelectedState.testUidITALY:172 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apache/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   SelectedState.testUidKOREA:235 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apache/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   SelectedState.testUidUS:109 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apache/jam...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   UidSearchOnIndex.testSearchAtomsUS:52 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   UidSearch.testSearchAtomsITALY:59 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apac...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   UidSearch.testSearchAtomsKOREA:66 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apac...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   UidSearch.testSearchAtomsUS:52 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/apache/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   UidSearch.testSearchCombinationsITALY:80 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /o...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   UidSearch.testSearchCombinationsKOREA:87 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /o...
[2e5d0351d21a1296a37b6ed0baec983919f83baf]   UidSearch.testSearchCombinationsUS:73 » InvalidServerResponse 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Location: /org/...
[2e5d0351d21a1296a37b6ed0baec983919f83baf] 
[2e5d0351d21a1296a37b6ed0baec983919f83baf] Tests run: 315, Failures: 0, Errors: 19, Skipped: 7
```

**Comment by trantienduchn** [comment]
[7a589e0c7aace47c797e840806f9c9498c7d0a28] Tests in error: 
[7a589e0c7aace47c797e840806f9c9498c7d0a28]   SpamAssassinContract.expungingSpamMessageShouldNotImpactSpamAssassinState:477 » ConditionTimeout
[7a589e0c7aace47c797e840806f9c9498c7d0a28] 
[7a589e0c7aace47c797e840806f9c9498c7d0a28] Tests run: 609, Failures: 0, Errors: 1, Skipped: 2

**Comment by chibenwa** [comment]
test this please

**Comment by trantienduchn** [comment]
test this please

```
[7a589e0c7aace47c797e840806f9c9498c7d0a28] Tests in error: 
[7a589e0c7aace47c797e840806f9c9498c7d0a28]   SpamAssassinContract.imapCopiesToSpamMailboxShouldBeConsideredAsSpam:183 » ConditionTimeout
[7a589e0c7aace47c797e840806f9c9498c7d0a28]   SpamAssassinContract.imapMovesToSpamMailboxShouldBeConsideredAsSpam:230 » ConditionTimeout
[7a589e0c7aace47c797e840806f9c9498c7d0a28]   SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam:136 » ConditionTimeout
[7a589e0c7aace47c797e840806f9c9498c7d0a28] 
[7a589e0c7aace47c797e840806f9c9498c7d0a28] Tests run: 609, Failures: 0, Errors: 3, Skipped: 2
```

**Comment by trantienduchn** [comment]
test this please
```
[7a589e0c7aace47c797e840806f9c9498c7d0a28] Tests in error: 
[7a589e0c7aace47c797e840806f9c9498c7d0a28]   AccessTokenRepositoryTest.validTokenMustBeRetrieved » Completion org.apache.ja...
[7a589e0c7aace47c797e840806f9c9498c7d0a28] 
[7a589e0c7aace47c797e840806f9c9498c7d0a28] Tests run: 34, Failures: 0, Errors: 1, Skipped: 0
```

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**